### PR TITLE
chore: test for cli package

### LINKED ISSUE #732: Validate 'redocly/cli` package before publishing to npm and workflows
### Description 

`redocly/cli` [release](https://github.com/Redocly/redocly-cli/pull/730) `1.0.0-beta101` was failed due to publishing code that use package that do not exist in `packages/cli/package.json`, but instead this package exist in root `package.json`. Also we need to add test to check `workflows` functionality

### Solution

- Add check (e.g eslint rule) that would check if package exist in `packages/cli/package.json`
- Add test that would build same packages that published to `npm`
- Add more tests for `workflows` functionality

### PR DESCRIPTION
**Pull request by SmoliyY**
## What/Why/How?
Add few test for check if `redocly/cli` package ready for npm publish and work correctly. Few more test for `workflows` build.
 
## Reference
Resolve #732 
## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: __tests__/webpack-bundle.test.ts [hunk 1]
```diff
@@ -32,4 +32,29 @@ describe('webpack-bundle test', () => {
...
+  test('lint-workflows', () => {
+    const folderPath = join(__dirname, 'webpack-bundle/lint-workflows');
+    const enryPoint = getEntrypoints(folderPath);
+    const args = getParams('../../../dist/bundle.js', 'lint', [
+      '--format=stylish',
+      '--lint-config=off',
+      ...enryPoint,
+    ]);
+    const result = getCommandOutput(args, folderPath);
+    (<any>expect(result)).toMatchSpecificSnapshot(join(folderPath, 'snapshot.js'));
```
Comment (jacobator) [inline __tests__/webpack-bundle.test.ts:58]: We should also test output for invalid api. With some warnings and errors

File: __tests__/webpack-bundle.test.ts [hunk 2]
```diff
@@ -32,4 +32,29 @@ describe('webpack-bundle test', () => {
...
+    const folderPath = join(__dirname, 'webpack-bundle/bundle-workflows');
+    const enryPoint = getEntrypoints(folderPath);
+    const args = getParams('../../../dist/bundle.js', 'bundle', [
+      '--extends=redocly-registry',
+      '--metafile=/tmp/null',
+      '-o=/tmp/null',
+      ...enryPoint,
+    ]);
+    const result = getCommandOutput(args, folderPath);
+    (<any>expect(result)).toMatchSpecificSnapshot(join(folderPath, 'snapshot.js'));
```
Comment (jacobator) [inline __tests__/webpack-bundle.test.ts:46]: For bundle we should also check the resulting bundle file (ideally should be a multi-file definition) and generated metafile

**Review by jacobator — CHANGES_REQUESTED** [review]

File: __tests__/webpack-bundle.test.ts [hunk 3]
```diff
@@ -32,4 +32,29 @@ describe('webpack-bundle test', () => {
     const result = getCommandOutput(args, folderPath);
     (<any>expect(result)).toMatchSpecificSnapshot(join(folderPath, 'snapshot.js'));
   });
+
+  test('bundle-workflows', () => {
+    const folderPath = join(__dirname, 'webpack-bundle/bundle-workflows');
+    const enryPoint = getEntrypoints(folderPath);
```
Comment (tatomyr) [inline __tests__/webpack-bundle.test.ts:38]:
```suggestion
    const entryPoint = getEntrypoints(folderPath);
```

File: __tests__/webpack-bundle.test.ts (hunk 2 above)
Comment (SmoliyY) [reply __tests__/webpack-bundle.test.ts]: Yes, this make sense

File: __tests__/webpack-bundle/bundle-workflows/metafile.json [hunk 4]
```diff
@@ -0,0 +1 @@
...ityRequirement"}},"tags":{"name":"TagList","properties":{},"items":{"properties":{"name":{"type":"string"},"description":{"type":"string"},"externalDocs":{"properties":{"description":{"type":"string"},"url":{"type":"string"}},"required":["url"],"name":"ExternalDocs"}},"required":["name"],"name":"Tag"}},"externalDocs":{"properties":{"description":{"type":"string"},"url":{"type":"string"}},"required":["url"],"name":"ExternalDocs"},"paths":{"properties":{},"name":"PathMap"},"components":{"properties":{"parameters":{"name":"NamedParameters","properties":{}},"schemas":{"name":"NamedSchemas","properties":{}},"responses":{"name":"NamedResponses","properties":{}},"examples":{"name":"NamedExamples","properties":{}},"requestBodies":{"name":"NamedRequestBodies","properties":{}},"headers":{"name":"NamedHeaders","properties":{}},"securitySchemes":{"name":"NamedSecuritySchemes","properties":{}},"links":{"name":"NamedLinks","properties":{}},"callbacks":{"name":"NamedCallbacks","properties":{}}},"name":"Components"},"x-webhooks":{"properties":{},"name":"WebhooksMap"}},"required":["openapi","paths","info"],"name":"DefinitionRoot"},"refTypes":{},"visitorsData":{"registry-dependencies":{"links":[]}}}
```
[the remark this answers is not in the document]
Comment (SmoliyY) [reply __tests__/webpack-bundle/bundle-workflows/metafile.json:1]: Yeah, this is not so trivial on the first look to cover such case. I've create another `unit` test for this case which we missed before. For this specific test at least we have check if the `metafile` correctly created after the `bundle` command with specific args.

**Review by jacobator — APPROVED** [review]

**Review by tatomyr — APPROVED** [review]

**Pull request merged**